agent/pyproject.toml

```
@@ -31,19 +31,64 @@ charm = [
     "pytest-operator>=0.42.0",
 ]
 dev = [
-    "black>=24.8.0",
-    "flake8>=5.0.4",
-    "pylint>=3.2.7",
     "pytest>=8.1.2",
     "pytest-cov>=5.0.0",
     "pytest-mock>=3.14.0",
     "pytest-timeout>=2.3.1",
     "pytest-xdist>=3.6.1",
     "requests-mock>=1.12.1",
+    "ruff>=0.11.6",
 ]
 
-[tool.black]
+[tool.ruff]
 line-length = 79
 
+[tool.ruff.lint]
+select = [
+    "A",   # flake8-builtins
+    "B",   # flake8-bugbear
+    "C4",  # comprenhension
+    "D",   # docstrings
+    "DTZ", # flake8-datetimez 
+    "E",   # pycodestyle errors
+    "F",   # pyflakes
+    "G",   # flake8-logging-format
+    "I",   # isort
+    "LOG", # flake8-logging
+    "N",   # naming
+    "PLE", # pylint errors
+    "PLW", # pyling warnings
+    "S",   # flake8-bandit 
+    "W",   # pycodestyle warnings
+]
+ignore = [
+    "D100", # docstring: Missing docstring in public module
+    "D101", # docstring: Missing docstring in public class
+    "D102", # docstring: Missing docstring in public method
+    "D103", # docstring: Missing docstring in public function
+    "D104", # docstring: Missing docstring in public package
+    "D107", # docstring: Missing docstring in __init__
+    "D205", # docstring: 1 black line required between summary line and description
+    "N803", # Naming: Argument '{name}' should be lowercase
+    "S108", # bandit: hardcoded-temp-file
```

Making a comment regarding this: I think it's okay to blanket ignore this for now since the agent code IS hardcoding to use `/tmp` for management purposes in a couple places.

I somewhat disagree with the fact it's hard-coded, but it is what it is, and we can ignore this error.